cargo/private/cargo_build_script_wrapper.bzl

```diff
@@ -107,6 +108,7 @@ def cargo_build_script(
             links attribute and therefore provide environment variables to this build script.
         proc_macro_deps (list of label, optional): List of rust_proc_macro targets used to build the script.
         build_script_env (dict, optional): Environment variables for build scripts.
```

Comment:
Wait, doesn't this get consumed as `cfg = "target"`? The order is:
```
rust_binary --exec--> cargo_build_script_runfiles --target--> cargo_build_script
```
If `env` if `build_script_env_for_target_cfg` is consumed through `cargo_build_script_runfiles` or directly on `cargo_build_script`, isn't the result the same?

Reply:
`cargo_build_script_runfiles` doesn't get `exec` transitioned - you put `cargo_build_script_wrapper` macros directly in `deps` (no edge transition), and that macro expands to a `cargo_build_script` which doesn't have a self transition. A `cargo_build_script` depends on a `cargo_build_script_runfiles` using the `script` attribute (which has `cfg = "target"`).

Reply:
But why is consuming `env` from `cargo_build_script_runfiles` different than consuming it directly in `cargo_build_script`?